lms/resources/frontend_app.py

```diff
@@ -0,0 +1,20 @@
+import functools
+
+from lms.resources._js_config import JSConfig
+
+
+class FrontendAppResource:
+    """
+    Generic context resource for views that serve frontend apps.
+
+    Certain frontend apps (eg. LTI launches) will have more specific context
+    resources.
+    """
+
+    def __init__(self, request):
+        self._request = request
+
+    @property
+    @functools.lru_cache()
+    def js_config(self):
+        return JSConfig(self, self._request)
```

Comment:
What happens if you do this instead? _If_ this works it's simpler

```suggestion
    def __init__(self, request):
        self.js_config = JSConfig(self, request)
```